.github/actions/build/action.yml

```diff
@@ -0,0 +1,26 @@
+name: build
```

Comment:
can we inline those actions in a dedicated zstd workflow for now? (and maybe later have a separate PR for refactoring the existing workflows w/ composite actions); I agree that we could have a better architecture of our workflows (maybe we could have a separate https://github.com/google/xls/issues/new?template=enhancement-proposal.yml for this?) but I think that it's orthogonal to the work being review here.